modules/table/src/main/java/org/apache/ignite/internal/table/distributed/replicator/PartitionReplicaListener.java

```diff
@@ -1944,7 +1944,7 @@ private <T> CompletableFuture<T> appendTxCommand(UUID txId, RequestType cmdType,
         });
 
         if (cleanupReadyFut.isCompletedExceptionally()) {
-            return failedFuture(new TransactionException(TX_FAILED_READ_WRITE_OPERATION_ERR, "Transaction is already finished."));
+            return failedFuture(new TransactionException(TX_ALREADY_FINISHED_ERR, "Transaction is already finished."));
```

Comment:
Is that true? Does it really means that if cleanupReadyFut was completed exceptionally then transaction was already finished?

Reply:
yes, see here

```
            if (txStateMeta == null || isFinalState(txStateMeta.txState())) {
                cleanupReadyFut.completeExceptionally(new Exception());
```

Reply:
yes, see this
```
            if (txStateMeta == null || isFinalState(txStateMeta.txState())) {
                cleanupReadyFut.completeExceptionally(new Exception());
```